src/gui/src/components/navigation/sidebar/sidebar-menu-link.tsx

```diff
@@ -0,0 +1,169 @@
+import { useState, Fragment } from 'react'
+import { motion, AnimatePresence } from 'framer-motion'
+import type { Variants } from 'framer-motion'
+import type { MenuItem } from '@/components/navigation/sidebar/menu.js'
+import {
+  SidebarMenuItem,
+  SidebarMenuButton,
+  SidebarMenuBadge,
+  SidebarMenuSub,
+  SidebarMenuSubItem,
+  SidebarMenuSubButton,
+  useSidebar,
+} from '@/components/ui/sidebar.jsx'
+import {
+  Collapsible,
+  CollapsibleContent,
+  CollapsibleTrigger,
+} from '@/components/ui/collapsible.jsx'
+import { useLocation, NavLink } from 'react-router'
+import { ArrowUpRight, ChevronRight } from 'lucide-react'
+import { cn } from '@/lib/utils.js'
+
+interface SidebarMenuLinkProps {
+  items: MenuItem[]
+  onParentClick?: () => void
+}
+
+const sublistVariants: Variants = {
+  hidden: { opacity: 0 },
+  visible: {
+    opacity: 1,
+    transition: {
+      staggerChildren: 0.05,
+    },
+  },
+}
+
+const sublistItemVariants: Variants = {
+  hidden: { opacity: 0, y: -1 },
+  visible: {
+    opacity: 1,
+    y: 0,
+    transition: { duration: 0.1, ease: 'easeInOut' },
+  },
+}
+
+export const SidebarMenuLink = ({
+  onParentClick,
+  items,
+}: SidebarMenuLinkProps) => {
+  const { pathname } = useLocation()
+  const { hoveredItem, setHoveredItem } = useSidebar()
+  const [openItems, setOpenItems] = useState<Record<string, boolean>>(
+    {},
+  )
+
+  const renderBadge = (item: MenuItem) => {
+    if (item.badge) {
+      return <SidebarMenuBadge>{item.badge}</SidebarMenuBadge>
+    }
+    return null
+  }
+
+  const renderItems = (items: MenuItem[], depth = 0) => {
```

Comment:
I believe both `renderBadge` and `renderItems` will be better off defined outside the main component. It might incur in some extra prop drilling but it avoids redefining these functions everytime the component is rendered.